mumax3c/scripts/energy.py

```diff
@@ -39,28 +38,39 @@ def zeeman_script(system):
     return mx3
 
 
-# Needs to be tidied up
-def uniaxialanisotropy_script(system):
+def uniaxialanisotropy_script(term, system):
     mx3 = "// UniaxialAnisotropy\n"
-    mx3 += f"Ku1 = {system.energy.uniaxialanisotropy.K1}\n"
-    # K2 to be added
-    mx3 += "anisu = vector({}, {}, {})\n\n".format(*system.energy.uniaxialanisotropy.u)
+    if not isinstance(term.K, ts.descriptors.Parameter):
+        mx3 += mc.scripts.set_parameter(parameter=term.K, name="Ku1", system=system)
+    else:
+        mx3 += mc.scripts.set_parameter(parameter=term.K1, name="Ku1", system=system)
+        mx3 += mc.scripts.set_parameter(parameter=term.K2, name="Ku2", system=system)
 
+    mx3 += mc.scripts.set_parameter(parameter=term.u, name="anisU", system=system)
     return mx3
 
 
-def demag_script(system):
+def demag_script(term, system):
     mx3 = "// Demag\n"
     mx3 += "enabledemag = true\n\n"
     return mx3
 
 
-def dmi_script(system):
-    mx3 = ""
-    if system.energy.dmi.crystalclass.lower() in ["t", "o"]:
-        name = "Dbulk"
-    elif system.energy.dmi.crystalclass.lower() == "cnv":
-        name = "Dind"
+def dmi_script(term, system):
+    if not system.energy.get(type=mm.Exchange):
+        raise ValueError(
+            "In mumax3 DMI cannot be used without exchange. "
+            "Solution: define exchange with a negligible A value."
+        )
+    elif system.energy.dmi.crystalclass.lower() in ["t", "o"]:
```

Comment:
We cannot use `system.energy.dmi` because we don't know if the term name is `dmi`.
```suggestion
    elif term.crystalclass.lower() in ["t", "o"]:
```